src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,789 @@
+    in if is_top varM then "⊤\n" else
+      match varM.d with
+      | None -> "⊥\n"
+      | Some arr -> if is_bot varM then "Bot \n" else Array.fold_left (fun acc elem -> acc ^ elem ) "" (Array.mapi show_var arr)
+
+  let pretty () (x:t) = text (show x)
+  let printXml f x = BatPrintf.fprintf f "<value>\n<map>\n<key>\nequalities-array\n</key>\n<value>\n%s</value>\n<key>\nenv\n</key>\n<value>\n%s</value>\n</map>\n</value>\n" (XmlUtil.escape (Format.asprintf "%s" (show x) )) (XmlUtil.escape (Format.asprintf "%a" (Environment.print: Format.formatter -> Environment.t -> unit) (x.env)))
+
+  let meet t1 t2 =
+    let sup_env = Environment.lce t1.env t2.env in
+    let t1, t2 = change_d t1 sup_env ~add:true ~del:false, change_d t2 sup_env ~add:true ~del:false in
+    let subst_var ts x t =
+      let adjust e = match e with
+        | (None, b') -> (None, b')
+        | (Some x', b') -> if x = x' then
+            (match t with
+             | (None, bt) -> (None, Z.(b' + bt))
+             | (Some xt, bt) -> (Some xt, Z.(b' + bt)))
+          else (Some x', b')in
+      Stdlib.Option.iter (BatArray.modify adjust) !ts
+    in
+    let add_conj ts t i =
+      let adjust ts' =
+        (match t with
+         | (None, b) ->
+           (match ts'.(i) with
+            | (None, b') -> if not @@ Z.equal b b' then ts := None;
+            | (Some j, b') -> subst_var ts j (None, Z.(b - b')))
+         | (Some j, b) ->
+           (match ts'.(i) with
+            | (None, b1) -> subst_var ts j (None, Z.(b1 - b))
+            | (Some h1, b1) ->
+              (match ts'.(j) with
+               | (None, b2) -> subst_var ts i (None, Z.(b2 + b))
+               | (Some h2, b2) ->
+                 if h1 = h2 then
+                   (if Z.(b1 <> (b2 + b)) then ts := None)
+                 else if h1 < h2 then subst_var ts h2 (Some h1, Z.(b1 - (b + b2)))
+                 else subst_var ts h1 (Some h2, Z.(b + (b2 - b1))))))
+      in
+      Stdlib.Option.iter adjust !ts
+    in
+    match t1.d, t2.d with
+    | Some d1', Some d2' -> (
+        let ds = ref (Some (Array.copy d1')) in
+        Array.iteri (fun j e -> add_conj ds e j) d2';
+        {d = !ds; env = sup_env} )
+    | _ -> { d = None; env = sup_env}
+
+  let meet t1 t2 =
+    let res = meet t1 t2 in
+    if M.tracing then M.tracel "meet" "meet a: %s b: %s -> %s \n" (show t1) (show t2) (show res) ;
+    res
+
+  let meet t1 t2 = timing_wrap "meet" (meet t1) t2
+
+  let leq t1 t2 =
+    let env_comp = Environment.compare t1.env t2.env in (* Apron's Environment.compare has defined return values. *)
+    let implies ts t i : bool =
+      match t with
+      | (None, b) ->
+        (match ts.(i) with
+         | (None, b') -> Z.equal b b'
+         | _ -> false)
+      | (Some j, b) ->
+        (match ts.(i), ts.(j) with
+         | (None, b1), (None, b2) -> Z.equal b1 (Z.add b2 b)
+         | (Some h1, b1), (Some h2, b2) ->
```

Comment:
Both cases could be combined with Option.equal (Int.equal) x y , or am I mistaken?
Also, ts.(i) is needed in both cases (line 389 and line 385) and could be done in a common let, while t could be deconstructed already as a parameter?